psi4/driver/driver_nbody.py

```diff
@@ -225,35 +206,116 @@ def nbody_gufunc(func, method_string, **kwargs):
     #    psioh = core.IOManager.shared_object()
     #    psioh.set_specific_retention(97, True)
 
-
     bsse_str = bsse_type_list[0]
     if len(bsse_type_list) >1:
         bsse_str =  str(bsse_type_list)
     core.print_out("\n\n")
     core.print_out("   ===> N-Body Interaction Abacus <===\n")
     core.print_out("        BSSE Treatment:                     %s\n" % bsse_str)
 
+    # Get compute list
+    compute_dict = build_nbody_compute_list(bsse_type_list, max_nbody, max_frag)
+
+    # Build size and slices dictionaries
+    fragment_size_dict = {frag: molecule.extract_subsets(frag).natom() for
+                                           frag in range(1, max_frag+1)}
+
+    # Compute N-Body components
+    e_dict, ptype_dict, int_dict = compute_nbody_components(func, method_string, molecule, compute_dict['all'], **kwargs)
```

Comment:
These function calls are quite complex. They either need to be simplified by duplicating small amounts of computation (broken out into functions) or a more complex data structure should be returned. Perhaps something like:

```
>>> build_nbody_compute_list(bsse_type_list, max_nbody, max_frag)
{
    "max_nbody": ...,
    "bsse_type_list": ...,
    ...
}

```

Reply:
To an extent, I believe that I had done that already for `build_nbody_compute_list()`; are you requesting similar data-structuring for the returned quantities of `compute_nbody_components()` and `assemble_nbody_components()`?  Should the arguments be `dict`-ified as well, for clarity?

Reply:
Effectively yes, but it must start with `build_nbody_compute_list` as this is the root that all methods will derive from. The `assemble_nbody_components` call is unwieldy enough that it likely is not very usable. I would suggest something like:

```
compute_metadata = build_nbody_compute_list(...)
component_results = compute_nbody_components(func, method_string, molecule, compute_metadata)
result = assemble_nbody_components(compute_metadata, component_results)
```

You may also want to consider making the last call simpler if `component_results` owns `compute_metadata`.

Reply:
Structure passing added.  For ease of driving w/in the `nbody_gufunc()` (so some things don't need recomputing) I've passed & returned the `metadata` dict itself between functions.  To call these functions independently, however, this isn't actually necessary -- so, I've documented in each function's docstring exactly _which_ `key: value` pairs in the `metadata` dictionary are required and/or added.  

I have no idea if Sphinx will scrape this properly -- but at @loriab's behest I've added an [item](https://github.com/orgs/psi4/projects/1#card-9213776) to the Documentation day topic list regarding standardization of the doc style for this structure passing.